### ISSUE TITLE
"[Cannot remove VM with detaching disks, VM is based from template.]"

### ISSUE THREAD

**Issue by bmanojlovic** [sinokylin/terraform-provider-ovirt#35]
when VM is created from template with existing base disks destroy would fail with this message
```
terraform destroy
data.ovirt_datacenters.DC: Refreshing state...
data.ovirt_storagedomains.sd: Refreshing state...
data.ovirt_networks.net02: Refreshing state...
data.ovirt_networks.net01: Refreshing state...
ovirt_vm.my_vm_1: Refreshing state... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc)
ovirt_disk.my_disk_1: Refreshing state... (ID: df9f67f6-19f0-4cdb-baa9-a5cd6e4cb916)
ovirt_disk_attachment.my_diskattachment_1: Refreshing state... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc:df9f67f6-19f0-4cdb-baa9-a5cd6e4cb916)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  - ovirt_disk.my_disk_1

  - ovirt_disk_attachment.my_diskattachment_1

  - ovirt_vm.my_vm_1


Plan: 0 to add, 0 to change, 3 to destroy.

Do you really want to destroy?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

ovirt_disk_attachment.my_diskattachment_1: Destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc:df9f67f6-19f0-4cdb-baa9-a5cd6e4cb916)
ovirt_disk_attachment.my_diskattachment_1: Destruction complete after 2s
ovirt_vm.my_vm_1: Destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc)
ovirt_disk.my_disk_1: Destroying... (ID: df9f67f6-19f0-4cdb-baa9-a5cd6e4cb916)
ovirt_disk.my_disk_1: Destruction complete after 0s
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 10s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 20s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 30s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 40s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 50s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 1m0s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 1m10s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 1m20s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 1m30s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 1m40s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 1m50s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 2m0s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 2m10s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 2m20s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 2m30s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 2m40s elapsed)
ovirt_vm.my_vm_1: Still destroying... (ID: 6e1e316e-614a-4c94-a6ec-76d419b585cc, 2m50s elapsed)

Error: Error applying plan:

1 error(s) occurred:

* ovirt_vm.my_vm_1 (destroy): 1 error(s) occurred:

* ovirt_vm.my_vm_1: Delete instalce timeout and got an error: Fault reason is "Operation Failed". Fault detail is "[Cannot remove VM with detaching disks, VM is based from template.]". HTTP response code is "409". HTTP response message is "409 Conflict".

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

**Comment by bmanojlovic** [comment]
API trace

```
DELETE /ovirt-engine/api/vms/6e1e316e-614a-4c94-a6ec-76d419b585cc?detach_only=true HTTP/1.1
Host: 127.0.0.1:8080
User-Agent: GoSDK/4.3.0.a1
Accept: application/xml
Authorization: Bearer XXX
Content-Type: application/xml
Version: 4
Connection: close

HTTP/1.1 409 Conflict
Date: Sat, 21 Jul 2018 13:19:49 GMT
Server: Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips
Content-Type: application/xml;charset=UTF-8
Content-Length: 200
Correlation-Id: 28eb36ba-af76-46ad-abf2-ffeac8a7e498
Connection: close

<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<fault>
    <detail>[Cannot remove VM with detaching disks, VM is based from template.]</detail>
    <reason>Operation Failed</reason>
</fault>
```

**Comment by imjoey** [comment]
Fixes by #36 .
